### PR TITLE
add show license template and tests

### PR DESCRIPTION
**Pull request by bobthebutcher**
##### ISSUE TYPE
 - New Template Pull Request

##### COMPONENT

- `cisco_ios_show_license`, 
- `Cisco IOS`
- `Show License`

##### SUMMARY
Add the show license template to capture license status on Cisco IOS devices.
Related to #440 

###### Before
```
Index 1 Feature: appxk9                         
        Period left: Life time
        License Type: Permanent
        License State: Active, In Use
        License Count: Non-Counted
        License Priority: Medium
Index 2 Feature: uck9                           
        Period left: Not Activated
        Period Used: 0  minute  0  second  
        License Type: EvalRightToUse
        License State: Active, Not in Use, EULA not accepted
        License Count: Non-Counted
        License Priority: None
Index 3 Feature: securityk9                     
        Period left: Life time
        License Type: Permanent
        License State: Active, In Use
        License Count: Non-Counted
        License Priority: Medium
Index 4 Feature: ipbasek9                       
        Period left: Life time
        License Type: Permanent
        License State: Active, In Use
        License Count: Non-Counted
        License Priority: Medium
Index 5 Feature: FoundationSuiteK9              
        Period left: Not Activated
        Period Used: 0  minute  0  second  
        License Type: EvalRightToUse
        License State: Active, Not in Use, EULA not accepted
        License Count: Non-Counted
        License Priority: None
Index 6 Feature: AdvUCSuiteK9                   
        Period left: Not Activated
        Period Used: 0  minute  0  second  
        License Type: EvalRightToUse
        License State: Active, Not in Use, EULA not accepted
        License Count: Non-Counted
        License Priority: None
Index 7 Feature: cme-srst                       
        Period left: Not Activated
        Period Used: 0  minute  0  second  
        License Type: EvalRightToUse
        License State: Active, Not in Use, EULA not accepted
        License Count: 0/0  (In-use/Violation)
        License Priority: None
Index 8 Feature: hseck9                         
        Period left: Life time
        License Type: Permanent
        License State: Active, In Use
        License Count: Non-Counted
        License Priority: Medium
Index 9 Feature: throughput                     
        Period left: Not Activated
        Period Used: 0  minute  0  second  
        License Type: EvalRightToUse
        License State: Active, Not in Use, EULA not accepted
        License Count: Non-Counted
        License Priority: None
Index 10 Feature: internal_service
```

###### After
```
[['appxk9',
  'Life time',
  '',
  'Permanent',
  'Active, In Use',
  'Non-Counted',
  'Medium'],
 ['uck9',
  'Not Activated',
  '0  minute  0  second  ',
  'EvalRightToUse',
  'Active, Not in Use, EULA not accepted',
  'Non-Counted',
  'None'],
 ['securityk9',
  'Life time',
  '',
  'Permanent',
  'Active, In Use',
  'Non-Counted',
  'Medium'],
 ['ipbasek9',
  'Life time',
  '',
  'Permanent',
  'Active, In Use',
  'Non-Counted',
  'Medium'],
 ['FoundationSuiteK9',
  'Not Activated',
  '0  minute  0  second  ',
  'EvalRightToUse',
  'Active, Not in Use, EULA not accepted',
  'Non-Counted',
  'None'],
 ['AdvUCSuiteK9',
  'Not Activated',
  '0  minute  0  second  ',
  'EvalRightToUse',
  'Active, Not in Use, EULA not accepted',
  'Non-Counted',
  'None'],
 ['',
  'Not Activated',
  '0  minute  0  second  ',
  'EvalRightToUse',
  'Active, Not in Use, EULA not accepted',
  '0/0  (In-use/Violation)',
  'None'],
 ['hseck9',
  'Life time',
  '',
  'Permanent',
  'Active, In Use',
  'Non-Counted',
  'Medium'],
 ['throughput',
  'Not Activated',
  '0  minute  0  second  ',
  'EvalRightToUse',
  'Active, Not in Use, EULA not accepted',
  'Non-Counted',
  'None']]
```

### CODE REVIEW

**Comment by bobthebutcher** [comment]
DOH, I noticed there is a key that is missing `cme-srst` Working on this now

**Comment by bobthebutcher** [comment]
> DOH, I noticed there is a key that is missing `cme-srst` Working on this now

Fixed by updating the Regex to match on non-whitespace chars

File: templates/cisco_ios_show_license.template [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+Value LICENSE_PRIORITY (.*)
+
+Start
+  ^Index\s\d\sFeature:\s${FEATURE}\s+$$
+  ^\s+Period left:\s${PERIOD_LEFT}$$
+  ^\s+Period Used:\s${PERIOD_USED}$$
+  ^\s+License Type:\s${LICENSE_TYPE}$$
+  ^\s+License State:\s${LICENSE_STATE}$$
+  ^\s+License Count:\s${LICENSE_COUNT}$$
+  ^\s+License Priority:\s${LICENSE_PRIORITY}$$ -> Record
```
Comment (FragmentedPacket) [inline templates/cisco_ios_show_license.template]: Please update to:
```
  ^\s*$$
  ^. -> Error
```

Also, anywhere in the state that is a space, can you change it to **\s*** or **\s+** depending on the case. It looks like most will be **\s**. Please do the same for all the regular **\s** as well.

Comment (bobthebutcher) [reply templates/cisco_ios_show_license.template]: Hey @FragmentedPacket im sorry for being dense, but I dont understand this part from your comments.
```
  ^\s*$$
  ^. -> Error
```
Can you talk me through this a bit more please ?

Comment (FragmentedPacket) [reply templates/cisco_ios_show_license.template]: We require each state (ex. **Start**) to end with those two lines. So you would add those after L16.

What it does is to make sure all lines are being accounted for from the output of the show command and any whitespace. 

Is that what you were looking for?

Comment (bobthebutcher) [reply templates/cisco_ios_show_license.template]: Ah yes, that is perfect thank you. I misunderstood the purpose and positioning, thank you very much for the explanation! I will get back with an update PR.

Comment (bobthebutcher) [reply templates/cisco_ios_show_license.template]: Thanks again for the help @FragmentedPacket I have made the updates now.

**Comment by bobthebutcher** [comment]
Alright found another edge case. Will need to investigate more and add some additional tests and adjust he pattern matching.

I found a device that has no properties for the `hsec` feature
```Index 8 Feature: hseck9```

I am thinking that licenses like this and the `internal_service` should be added to the result with empty property values.

**Comment by FragmentedPacket** [comment]
@bobthebutcher Sounds good. Let me know if you need any help.

**Comment by FragmentedPacket** [comment]
@bobthebutcher Bump on this

**Comment by bobthebutcher** [comment]
Apologies, I have been flat out with other work and have not had time to get back to this. BUT! I will be freed up at the end of the month so will come back to this towards the end of next week.

**Comment by jmcgill298** [comment]
#533
